### PR TITLE
[IMP] point_of_sale: disallow sudo commands without password

### PR DESCRIPTION
**Pull request by loouis-t**
Currently, if one connects to `pi` user with ssh, password is not required (after connection) to run sudo commands.

Now, password is required for sudo commands with `pi` user, and Odoo service uses a new `odoo` user we can't connect to (that doesn't require password for sudo commands).

Task: 4345733

### CODE REVIEW

**Comment by loouis-t** [comment]
need to make `odoo` user path: `/home/pi`

File: addons/hw_drivers/tools/helpers.py [hunk 1]
```diff
@@ -476,7 +476,7 @@ def odoo_restart(delay=0):
 def path_file(filename):
     platform_os = platform.system()
     if platform_os == 'Linux':
-        return Path.home() / filename
+        return Path("/home/pi") / filename  # Path.home() returns odoo user's home instead of pi's
```
Comment (MaxTheWhale) [inline addons/hw_drivers/tools/helpers.py:479]: There are also more usages of this in `KeyboardUSBDriver_L.py` (I guess they can change to use this helper function)

**Comment by MaxTheWhale** [comment]
I've built an image using this and it seems to be working perfectly :)

File: addons/hw_drivers/tools/helpers.py (hunk 1 above)
Comment (loouis-t) [reply addons/hw_drivers/tools/helpers.py:479]: right, thank's, it's updated !

**Comment by loouis-t** [comment]
@odoo/rd-security could you have a look at this please?

## Context

Currently, the odoo service on IoT Boxes is runned by `pi` user.
This user is also used by people connecting to the rpi via ssh.

### Solution proposed

We created `odoo` user, that has same permissions as pi except we can't connect to it.
This allows us granting sudo permissions to `odoo` without asking for password, and require password only for `pi` user.
**Note**: we can't use a system user instead of regular one (and thus avoid creating a home directory for it) as odoo service has to be able to log in a `lightdm` session (not bash one) to control displays.

## Security issues

- `return Path("/home/pi") / filename`
  - **Reason for the use:** code is in `/home/pi`, but `odoo` user's home is `/home/odoo`
  - **Safe because:** base path is harcoded.

**Review by xmo-odoo — COMMENTED** [review]
@robodoo override=ci/security

**Comment by loouis-t** [comment]
**note to self:** some handlers seem to be owned by root, despite we chowned them to `odoo:odoo`

**Review by Ysoroko — CHANGES_REQUESTED** [review]
everything works fine except for the `odoo` shortcut:

```
pi@raspberrypi:~/odoo/addons/hw_drivers $ odoo
Traceback (most recent call last):
  File "/home/pi/odoo/odoo-bin", line 6, in <module>
    odoo.cli.main()
  File "/home/pi/odoo/odoo/cli/command.py", line 106, in main
    o.run(args)
  File "/home/pi/odoo/odoo/cli/server.py", line 180, in run
    main(args)
  File "/home/pi/odoo/odoo/cli/server.py", line 170, in main
    setup_pid_file()
  File "/home/pi/odoo/odoo/cli/server.py", line 93, in setup_pid_file
    with open(config['pidfile'], 'w') as fd:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/run/odoo/odoo.pid'
```

**Review by Ysoroko — APPROVED** [review]
@robodoo r+

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**